source/common/http/conn_manager_utility.cc

```diff
@@ -445,11 +447,10 @@ void ConnectionManagerUtility::mutateXfccRequestHeader(RequestHeaderMap& request
   }
 }
 
-void ConnectionManagerUtility::mutateResponseHeaders(ResponseHeaderMap& response_headers,
-                                                     const RequestHeaderMap* request_headers,
-                                                     ConnectionManagerConfig& config,
-                                                     const std::string& via,
-                                                     bool clear_hop_by_hop) {
+void ConnectionManagerUtility::mutateResponseHeaders(
+    ResponseHeaderMap& response_headers, const RequestHeaderMap* request_headers,
+    ConnectionManagerConfig& config, const std::string& via,
+    const StreamInfo::StreamInfo& stream_info, absl::string_view node_id, bool clear_hop_by_hop) {
```

Comment:
Instead of passing `node_id` around, why not passing cached proxy name? If we have more options for proxy name this list doesn't grow then.

Reply:
Could you point me to the cached proxy name value? Sorry, I was unable to find it.

Reply:
I meant you can pre-calculate it (and store in HCM) instead of passing `node_id` around.

Reply:
Thanks, this is a really excellent idea. Implemented in 071c406, ptal.